hpx/runtime/threads/detail/scheduling_loop.hpp

```diff
@@ -933,6 +933,12 @@ namespace hpx { namespace threads { namespace detail
                     params.inner_();
             }
 
+            if (scheduler.SchedulingPolicy::get_scheduler_mode() &
+                    policies::enable_user_polling)
+            {
+                scheduler.user_polling_function();
+            }
```

Comment:
Shouldn't this better go to the background thread instead?

Reply:
The background thread runs as an hpx task and can be suspended. Since we're only calling MPI on this and it pretty much expects to be on a std::thread type thread, I'm not sure that it really mattters where we call it. For this initial implementation I put it outside the background threading work to ensure I know what's going on, however it could be moved into the background work once I'm happy that all is behaving sensibly. (does it really make any differnce where we do it? what are the pros and cons?)

Reply:
Well, cons are that the current implementation adds overheads to the inner part of the scheduling loop for everything, not just for your MPI executor.

Reply:
See also #4507